### PR TITLE
Pin migrate binary

### PR DESCRIPTION
**Pull request by riyazdf**
temporary workaround for https://github.com/mattes/migrate/issues/202 since we have not tested v3 nor does it support all drivers yet

Closes https://github.com/docker/notary/issues/1142

Signed-off-by: Riyaz Faizullabhoy <riyaz.faizullabhoy@docker.com>

### CODE REVIEW

File: server.Dockerfile [hunk 1]
```diff
@@ -3,8 +3,10 @@ MAINTAINER David Lawrence "david.lawrence@docker.com"
 
 RUN apk add --update git gcc libc-dev && rm -rf /var/cache/apk/*
 
-# Install SQL DB migration tool
-RUN go get github.com/mattes/migrate
+# Pin to the specific v1 version
+RUN git clone -b v1 https://github.com/mattes/migrate.git /go/src/github.com/mattes/migrate/
+RUN go get -u -v github.com/mattes/migrate && \
```
Comment (endophage) [inline server.Dockerfile]: Fewer layers (for all that it matters in the weeks until we get 17.05 multi-stage builds):
```
RUN git clone ... \
    && go get ...
```
Also, not sure we need the `-u` flag. If we just cloned, it shoudn't be necessary to ask go to also do a network update.

Comment (endophage) [inline server.Dockerfile]: Also, if it's already cloned, you might just be able to do a `go install -tags 'mysql' =o ...` and skip the `go get` entirely

Comment (riyazdf) [inline server.Dockerfile]: Removed `-u` flag and collapsed into one layer.

I tried skipping the `go get` and ran into dependency issues on the build, so left as is:
```
Cloning into '/go/src/github.com/mattes/migrate'...
src/github.com/mattes/migrate/driver/ql/ql.go:13:2: cannot find package "github.com/cznic/ql/driver" in any of:
	/usr/local/go/src/github.com/cznic/ql/driver (from $GOROOT)
	/go/src/github.com/cznic/ql/driver (from $GOPATH)
...
```

Comment (endophage) [inline server.Dockerfile]: Ah yeah, that makes sense if migrate isn't vendoring dependencies (which it isn't)

**Comment by riyazdf** [comment]
jenkins needed a kick, repushed the same commit.

**Comment by cyli** [comment]
LGTM!  Thanks for fixing this so quickly!

**Pull request merged**